### PR TITLE
Add 100Mi of overhead for guaranteed vmis

### PR DESCRIPTION
**Pull request by fossedihelm**
**What this PR does / why we need it**:
Preliminary tests showed that the virt-overhead is underestimated by 100Mi. For guaranteed vmi this means that the limits we set are too stringent.

Also, add the full unit test coverage of the `getMemoryOverhead` function.
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:

```release-note
NONE
```

/cc @xpivarc

### CODE REVIEW

File: pkg/virt-controller/services/renderresources_test.go [hunk 1]
```diff
@@ -431,6 +434,301 @@ var _ = Describe("Resource pod spec renderer", func() {
...
+		}
+		staticOverhead = resource.NewScaledQuantity(0, resource.Kilo)
+		staticOverhead.Add(resource.MustParse(VirtLauncherMonitorOverhead))
+		staticOverhead.Add(resource.MustParse(VirtLauncherOverhead))
+		staticOverhead.Add(resource.MustParse(VirtlogdOverhead))
+		staticOverhead.Add(resource.MustParse(VirtqemudOverhead))
+		staticOverhead.Add(resource.MustParse(QemuOverhead))
+		// static overhead for IOThread
+		staticOverhead.Add(resource.MustParse("8Mi"))
+
```
Comment (xpivarc) [inline pkg/virt-controller/services/renderresources_test.go:475]: I think you can simplify and just use constant value. This feels like duplicating the production logic

**Review by xpivarc — COMMENTED** [review]
/approve
The approach is good, but I am not sure about the tests.

**Comment by fossedihelm** [comment]
> /approve The approach is good, but I am not sure about the tests.

Thanks! I pushed some changes in the direction you suggested, PTAL!
Also, I fixed a typo and aligned the behavior of initializing the vmi in the `BeforeEach`. Thanks

**Comment by fossedihelm** [comment]
/hold
Waiting for https://github.com/kubevirt/kubevirt/pull/10571 to be merged first

**Comment by brianmcarey** [comment]
/hold

sig-performance presubmit has failed a number of times - hold to stop retest.

File: pkg/virt-controller/services/renderresources.go [hunk 2]
```diff
@@ -404,6 +404,10 @@ func GetMemoryOverhead(vmi *v1.VirtualMachineInstance, cpuArch string, additiona
 		overhead = multiplyMemory(overhead, ratio)
 	}
 
+	if vmi.IsCPUDedicated() || vmi.WantsToHaveQOSGuaranteed() {
+		overhead.Add(resource.MustParse("100Mi"))
```
Comment (fabiand) [inline pkg/virt-controller/services/renderresources.go:408]: I wonder if we should bump this to 200MiB

Main reason being: It gives the VM more space for buffers.
A lack of buffers can lead to performance issues.

But - I've got no good data to support this statement.

Comment (xpivarc) [reply pkg/virt-controller/services/renderresources.go:408]: The same goes for 300Mi, could be even better.
We based the 100Mi base on what we saw in our test so far. I think fine-tuning requires a more holistic approach for this.

**Comment by rthallisey** [comment]
@fossedihelm this needs a release note that mem increased by 100m in certain cases.

**Comment by fossedihelm** [comment]
@rthallisey Thank you! I edited the release note section. Should it works? Thanks

**Pull request merged**